packages/mui-material/src/Link/Link.js

```diff
@@ -149,7 +150,7 @@ const Link = React.forwardRef(function Link(inProps, ref) {
     ...props,
     // It is too complex to support any types of `sx`.
     // Need to find a better way to get rid of the color manipulation for `textDecorationColor`.
-    color: (typeof sx === 'function' ? sx(theme).color : sx?.color) || color,
+    color: (typeof sxColor === 'function' ? sxColor(theme) : sxColor) || color,
```

Comment:
IMHO

```jsx
      <Link href="#unknown" color="#ff5252">
        #ff5252
      </Link>
```

and

```jsx
      <Link href="#unknown" sx={{ color: '#ff5252' }}>
        #ff5252
      </Link>
```

should behave differently. I would expect the former to set the color business logic of the component, e.g. the underline color, while the latter to set the CSS color. I think that this would be correct:

```suggestion
    color,
```

Based on what I can see in https://www.argos-ci.com/mui/material-ui/builds/400. Anyway, nothing major, It's more that I feel we are overstepping a bit the scope responsibility of the system.

Reply:
Agree, I think it is too soon to add it. `color` in `sx` should only target CSS color property. I can open a PR to revert it until it is too late. @mnajdova What do you think?

Reply:
Agree, the keys inside the `sx` prop should correspond to the CSS properties only